src/GafferScene/BranchCreator.cpp

```diff
@@ -321,7 +321,11 @@ IECore::ConstInternedStringVectorDataPtr BranchCreator::computeChildNames( const
 
 void BranchCreator::hashSetNames( const Gaffer::Context *context, const ScenePlug *parent, IECore::MurmurHash &h ) const
 {
-	ConstCompoundDataPtr mapping = boost::static_pointer_cast<const CompoundData>( mappingPlug()->getValue() );
+	ConstCompoundDataPtr mapping;
+	{
+		ScenePlug::GlobalScope s( context );
```

Comment:
Hmm. Something seems wrong here. Anything pulling on the setNames plug should be using a global context already - that's why there's a `ScenePlug::setNamesHash()` method. By putting extra context management in here, we're penalising all the caller's who've done the right thing. It seems highly likely that we have some rogue code elsewhere, and we should be optimising that instead.

Where I _would_ think a GlobalScope might help would be where we expect "scene:path" to be in the context, but actually I think I've mocked that up in the past, and not seen any gains, at least for the things I was testing at the time.